File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let curString = '';
    let maxLength = 0;
    
    for(let i = 0; i < s.length; i++) {
        
        let curLetter = s[i];
        let indexOfCurStr = curString.indexOf(curLetter);
        
        if(indexOfCurStr > -1) {
            curString = curString.substring(indexOfCurStr + 1);
        }
        
        curString += curLetter;
        maxLength = Math.max(maxLength, curString.length);

    }
    return maxLength;
};

File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start = maxLength = 0
        curString = {}
        
        for i in range(len(s)):
            if s[i] in curString and start <= curString[s[i]]:
                start = curString[s[i]] + 1
            else:
                maxLength = max(maxLength, i - start + 1)
                
            curString[s[i]] = i
            
        return maxLength
        